solver/src/solver.rs

```diff
@@ -15,6 +15,9 @@ use reqwest::Url;
 use shared::{price_estimate::PriceEstimating, token_info::TokenInfoFetching};
 use structopt::clap::arg_enum;
 
+/// Defines shared behaviour for all Solvers.
```

Comment:
It's really the interface that all solvers need to implement (there is no behavior in this trait definition).